pkg/kubectl/cmd/get.go

```diff
@@ -248,6 +248,22 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 	sorting, err := cmd.Flags().GetString("sort-by")
 	var sorter *kubectl.RuntimeSort
 	if err == nil && len(sorting) > 0 && len(objs) > 1 {
+
+		clientConfig, err := f.ClientConfig()
+		if err != nil {
+			return err
+		}
+
+		version, err := cmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
+		if err != nil {
+			return err
+		}
+
+		objs, err = resource.AsVersionedObjects(infos, version.String(), f.JSONEncoder())
+		if err != nil {
+			return err
+		}
+
 		// TODO: questionable
 		if sorter, err = kubectl.SortObjects(f.Decoder(true), objs, sorting); err != nil {
```

Comment:
I would expect us to be sorting based on the JSON and the field path
qualifier only, or at minimum on the map[string]interface{}.  It's a
riskier change though.

On Mon, Feb 22, 2016 at 1:37 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/get.go
> <https://github.com/kubernetes/kubernetes/pull/21694#discussion_r53669745>
> :
>
> > +
> > +		clientConfig, err := f.ClientConfig()
> > +		if err != nil {
> > +			return err
> > +		}
> > +
> > +		version, err := cmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
> > +		if err != nil {
> > +			return err
> > +		}
> > +
> > +		objs, err = resource.AsVersionedObjects(infos, version.String(), f.JSONEncoder())
> > +		if err != nil {
> > +			return err
> > +		}
> > +
> >  		// TODO: questionable
> >  		if sorter, err = kubectl.SortObjects(f.Decoder(true), objs, sorting); err != nil {
>
> at first glance, rather than converting into versions objects, here, I
> *think* would expect to pass the output version to SortObjects... it is
> taking a fieldInput, which is not really meaningful unless paired with a
> group version. In SortObjects, can we decode to the specified
> groupversion? I'd like @smarterclayton <https://github.com/smarterclayton>'s
> take on that as well...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/21694/files#r53669745>.
>